solver/src/driver.rs

```diff
@@ -323,7 +327,10 @@ impl Driver {
             .await
     }
 
-    pub async fn single_run(&mut self) -> Result<()> {
+    pub async fn single_run(
+        &mut self,
+        inflight_orders: &HashSet<OrderUid>,
```

Comment:
I'm not entirely sure the driver of this method should manage state about inflight orders. While this approach makes it very clear that inflight orders only stick around for the next run, I wonder if storing them as a field on the driver instance itself would be more contained.